File: java/src/main/java/problems201_300/DeleteNodeInALinkedList.java
package problems201_300;

import concepts.SinglyListNode;

enum DeleteNodeInALinkedList {
    DEFAULT {
        @Override
        void deleteNode(SinglyListNode node) {
            if (node == null) {
                return;
            }
            SinglyListNode next = node.next;
            node.val = next.val;
            node.next = next.next;
        }
    };

    abstract void deleteNode(SinglyListNode node);
}

File: kotlin/src/main/kotlin/problems201_300/DeleteNodeInALinkedList.kt
package problems201_300

import concepts.SinglyListNode

enum class DeleteNodeInALinkedList {
    DEFAULT {
        override fun deleteNode(node: SinglyListNode?) {
            if (node == null) {
                return
            }
            val next = node.next
            node.`val` = next!!.`val`
            node.next = next.next
        }
    },
    ;

    abstract fun deleteNode(node: SinglyListNode?)
}
